### ISSUE TITLE
fig.colorbar() and using bbox='tight' in PDF export mess up figure dimensions

### ISSUE THREAD

**Issue by quantsnus** [matplotlib/matplotlib#18165]
### Bug report

**Bug summary**

After upgrading from matplotlib 3.2.2 to 3.3.0 a previously nicely working plot was messed up in it's dimensions location and depending on the figure dimensions parts could be cropped in the output. 

I reduced the plotting to an example which should show, that this has something to do with adding a colorbar using fig.colorbar() and exporting to PDF applying the bbox='tight' option. If either of the two is left out the output is as expected.

**Code for reproduction**

```python
import matplotlib.pyplot as plt
fig, axarr = plt.subplots(2,1, figsize=(8,10))
im0 = axarr[0].imshow([[0,1],[1,0]], aspect='auto')
im1 = axarr[1].imshow([[0,1],[1,0]], aspect='auto')
colorBar=fig.colorbar(im0, ax=axarr)
plt.savefig('test.png', bbox_inches='tight')
plt.savefig('test.pdf', bbox_inches='tight')
plt.savefig('test2.png', bbox_inches='tight')
```

**Actual outcome**

test2.png is output significantly cropped
![test2](https://user-images.githubusercontent.com/25102059/89274700-8fae3b00-d641-11ea-8fe7-f60ea502a88f.png)

**Expected outcome**

test.png comes out as expected
![test](https://user-images.githubusercontent.com/25102059/89274711-9472ef00-d641-11ea-96b5-92fe7615184f.png)

For completeness also the pdf output, which is as expected:
[test.pdf](https://github.com/matplotlib/matplotlib/files/5021137/test.pdf)

**Matplotlib version**

  * Operating system: Windows 10
  * Matplotlib version: 3.3.0
  * Matplotlib backend: Qt5Agg
  * Python version: 3.7.8
  * Jupyter version (if applicable): -
  * Other libraries: -

Using conda, installed matplotlib from conda-forge

**Comment by jklymak** [comment]
This should be fixed in #18089

**Comment by jklymak** [comment]
... and will be fixed in 3.3.1
